src/TraceEvent/TraceEvent.cs

```diff
@@ -2816,6 +2816,28 @@ public virtual void StopProcessing()
         /// Subscribers of Completed will be called after processing is complete (right before TraceEventDispatcher.Process returns.    
         /// </summary>
         public event Action Completed;
+
+        /// <summary>
+        /// Add a hook right before dispatch that determines if an event should be dispatched
+        /// Multiple subscribers may hook this event, in that case they are called in lifo order,
+        /// once a subscriber indicates that an event should not be processed the chain of dispatchers is no longer searched
+        /// <input>TraceEvent - the event that is ready for dispatch</input>
+        /// <return>True - dispatch this event</return>
+        /// <return>False - not dispatch the event</return>
+        /// </summary>
+        public void AddDispatchHook(Func<TraceEvent, bool> onDispatch)
+        {
+            var hook = new DispatchHookEntry()
+            {
+                OnDispatch = onDispatch,
+                Next = null
+            };
+
+            // last in first out order
+            if (DispatchHook != null) hook.Next = DispatchHook;
```

Comment:
:bulb: This null check is unnecessary. You could just initialize `Next` to `DispatchHook` from the start.